plugins/common/opcua/client.go

```diff
@@ -44,7 +44,8 @@ type OpcUAClientConfig struct {
 	ConnectTimeout config.Duration `toml:"connect_timeout"`
 	RequestTimeout config.Duration `toml:"request_timeout"`
 
-	Workarounds OpcUAWorkarounds `toml:"workarounds"`
+	OptionalFields []string         `toml:"optional_fields"`
```

Comment:
Could you please check if all parameters given for `optional_fields` are valid values? Currently, you silently ignore the option on misspellings like `optional_fields = ["datatype"]` or `optional_fields = ["DataTpye"]` which might be surprising to the user. I guess `func (o *OpcUAClientConfig) Validate() error` would be a good place to check.

Reply:
Implemented